api/src/middlewares/services/UserService/personalisation/personalisationBuilder.userConfirmation.ts

```diff
@@ -4,17 +4,18 @@ import { AnswersHashMap } from "../../../../types/AnswersHashMap";
 import { FormType, PayMetadata } from "../../../../types/FormDataBody";
 import { ApplicationError } from "../../../../ApplicationError";
 
-const previousMarriageDocs = {
-  Divorced: "decree absolute",
-  "Dissolved civil partner": "final order",
-  Widowed: "late partner's death certificate",
-  "Surviving civil partner": "late partner's death certificate",
-  Annulled: "decree of nullity",
+const personalisationTypeMap = {
+  affirmation: getAffirmationPersonalisations,
+  cni: getCNIPersonalisations,
 };
 
 export function buildUserConfirmationPersonalisation(answers: AnswersHashMap, metadata: { reference: string; payment?: PayMetadata; type?: FormType }) {
   const isSuccessfulPayment = metadata.payment?.state?.status === "success" ?? false;
-  const docsList = buildUserConfirmationDocsList(answers, metadata.type);
+
+  if (!answers) {
+    throw new ApplicationError("WEBHOOK", "VALIDATION", 500, "Fields are empty");
+  }
+  const additionalPersonalisations = personalisationTypeMap[metadata.type as string]?.(answers) ?? {};
```

Comment:
We can leverage the fact that this is typed, rather than casting it to string. I assume the others are just not implemented yet? We can just return empty objects in this case

```.ts
type AdditionalPersonalisationsByFormType = Record<FormType, (answers: AnswersHashMap) => { [key: string]: string | string[] | boolean }>;

function getAffirmationPersonalisation(answers: AnswersHashMap) {
  return {
    //...
  };
}

const getPersonalisationByFormType: AdditionalPersonalisationsByFormType = {
  affirmation: getAffirmationPersonalisation,
  cni(answers: AnswersHashMap) {
    return {};
  },
  exchange(answers: AnswersHashMap) {
    return {};
  },
  cniAndMsc(answers: AnswersHashMap) {
    return {};
  },
  msc(answers: AnswersHashMap) {
    return {};
  },
};

getPersonalisationByFormType.affirmation(answers)
// or do an additional check for the presence of the function
const getAdditionalPersonalisations = getPersonalisationByFormType[metadata.type];
if(!getAdditionalPersonlisations) {
 // log/throw or continue
}


```